source/module_io/get_wf_lcao.cpp

```diff
@@ -475,27 +415,219 @@ void IState_Envelope::begin(const psi::Psi<std::complex<double>>* psi,
+                    std::vector<double> wfc_real(wfcpw->nrxx);
+                    std::vector<double> wfc_imag(wfcpw->nrxx);
+                    for (int ir = 0; ir < wfcpw->nrxx; ++ir)
+                    {
+                        wfc_real[ir] = wfc_r[ir].real();
+                        wfc_imag[ir] = wfc_r[ir].imag();
+                    }
+
+                    // Output real part
+                    std::stringstream ss_real;
+                    ss_real << global_out_dir << "BAND" << ib + 1 << "_k_" << ik + 1 << "_s_" << ispin + 1
+                            << "_REAL.cube";
+                    const double ef_tmp = this->pes_->eferm.get_efval(ispin);
+                    ModuleIO::write_cube(
+#ifdef __MPI
+                        bigpw->bz,
+                        bigpw->nbz,
+                        rhopw->nplane,
+                        rhopw->startz_current,
+#endif
+                        wfc_real.data(),
+                        ispin,
+                        nspin,
+                        0,
+                        ss_real.str(),
+                        rhopw->nx,
+                        rhopw->ny,
+                        rhopw->nz,
+                        ef_tmp,
+                        &(GlobalC::ucell));
+
+                    // 输出虚部
+                    std::stringstream ss_imag;
+                    ss_imag << global_out_dir << "BAND" << ib + 1 << "_k_" << ik + 1 << "_s_" << ispin + 1
+                            << "_IMAG.cube";
+                    ModuleIO::write_cube(
+#ifdef __MPI
+                        bigpw->bz,
+                        bigpw->nbz,
+                        rhopw->nplane,
+                        rhopw->startz_current,
+#endif
+                        wfc_imag.data(),
+                        ispin,
+                        nspin,
+                        0,
+                        ss_imag.str(),
+                        rhopw->nx,
+                        rhopw->ny,
+                        rhopw->nz,
+                        ef_tmp,
+                        &(GlobalC::ucell));
+                }
+            }
+        }
     }
 
     for (int ik = 0; ik < nks; ++ik)
     {
-        for (int ib = 0; ib < nbands; ++ib) {
+        for (int ib = 0; ib < nbands; ++ib)
+        {
             delete[] wfc_k_grid[ik][ib];
-}
+        }
         delete[] wfc_k_grid[ik];
     }
 
     return;
 }
 
+void IState_Envelope::select_bands(const int nbands_istate,
+                                   const std::vector<int>& out_wfc_kb,
+                                   const int nbands,
+                                   const double nelec,
+                                   const int mode,
+                                   const int fermi_band)
+{
+    ModuleBase::TITLE("IState_Envelope", "select_bands");
+
+    int bands_below = 0;
+    int bands_above = 0;
+
+    this->bands_picked_.resize(nbands);
+    ModuleBase::GlobalFunc::ZEROS(bands_picked_.data(), nbands);
+
+    if (mode == 1)
```

Comment:
add some annotation about what is mode 1? Also for line 551, what is mode 2, thanks.